### ISSUE TITLE
[Bug]: Type error: Type instantiation is excessively deep and possibly infinite.

### ISSUE THREAD

**Issue by RomkeSandwave** [puppeteer/puppeteer#9369]
### Bug description

Steps to reproduce the problem:

1. Query a complex selector in a Typescript puppeteer test script
2. Typescript complains that resolving the return type takes too many steps: `Type error: Type instantiation is excessively deep and possibly infinite.`

### Puppeteer version

19.3.0

### Node.js version

16.18.1

### npm version

yarn 1.22.19

### What operating system are you seeing the problem on?

Linux

### Relevant log output

```Shell
Type error: Type instantiation is excessively deep and possibly infinite.

  70 | 
> 71 |     const rows = await page.$$( "[data-testid=\"my-component\"] tbody tr" )
     |                        ^
```

**Comment by OrKoN** [comment]
@RomkeSandwave which typescript version do you have?

**Comment by RomkeSandwave** [comment]
@OrKoN 4.4.3

I'll try upgrading to 4.9.3 and see if that solves it.

**Comment by RomkeSandwave** [comment]
Looks to be fixed in Typescript 4.9.3.
